library/src/main/java/com/hannesdorfmann/adapterdelegates4/AdapterItemProvider.java

```diff
@@ -0,0 +1,11 @@
+package com.hannesdorfmann.adapterdelegates4;
+
+import androidx.annotation.Nullable;
+
+public interface AdapterItemProvider<T> {
```

Comment:
When I started with this library I was thinking about introducing something like this but then I decided to not do that as it felt like reinventing the Adapter class (which has that API already but Adapter class in general has no clear single responsibility this is where my motivation for introducing something like this came from). 

I then thought it would be easier to let people subclass adapter and override the corresponding methods. What do you think? Do I miss something?

Reply:
The point is that AdapterDelegate has access to "datasource", which is by default is List in the library.
I don't know whether accessing datasource from delegate is really something that is needed in API. Seems I haven't used it.
But if it is needed, then for me it seems that List is not the best default option. Default option should be some "random-access" interface.
List provides such access. Adapter provides such access.
But not all adapters have List as a "datasource".

That is why I think some general interface most likely should be added. Because List is too specific and limits for example usage with Paginated adapters, which don't have List under the hood.

We could pass Adapter into Delegate directly. But yes, Adapter has too many responsibilities, so it feels it should be hidden under some interface.

Naming is the question though.
Maybe "AdapterDatasource", "DelegationDatasource" or so could be better than AdapterItemProvider